### PR TITLE
[Serializer] improve documentation of default_context for the serializer

### LINKED ISSUE #17306: Add context option to configure the indentation of nested nodes for `YamlEncoder`
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#47243
| PR author(s) | @dbu
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by dbu**
additional info for https://github.com/symfony/symfony-docs/pull/17310

fix #17306

the code example will only work once https://github.com/symfony/symfony/pull/47243 is merged

### CODE REVIEW

**Comment by dbu** [comment]
build should get green when retrying, as the code is now merged to 6.2.

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -2743,6 +2743,18 @@ paths
...
+...............
+
+**type**: ``array`` **default**: ``[]``
+
+A map with default context options that will be used with each ``serialize`` and ``deserialize``
+call. This can be used for example to set the json encoding behavior by setting ``json_encode_options``
+to a `json_encode flags bitmask`_.
+
+You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
+to discover the available settings.
```
Comment (OskarStark) [inline reference/configuration/framework.rst:2756]:
```suggestion
to discover the available settings.

.. versionadded:: 6.2

    The ``default_context`` parameter was introduced in Symfony 6.2.
```

**Review by OskarStark — APPROVED** [review]

File: reference/configuration/framework.rst (hunk 1 above)
Comment (dbu) [reply reference/configuration/framework.rst:2756]: that is not correct. default_context has been added in symfony 5.4 but was not documented aparently. 
https://github.com/symfony/symfony/blob/5.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1010

Comment (dbu) [reply reference/configuration/framework.rst:2756]: the context builders seem to have been added in 6.1 https://github.com/symfony/symfony/tree/6.1/src/Symfony/Component/Serializer/Context

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -2743,6 +2743,22 @@ paths
...
+
+.. versionadded:: 6.2
```
Comment (dbu) [inline reference/configuration/framework.rst]:
```suggestion
.. versionadded:: 5.4
```

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -2743,6 +2743,22 @@ paths
...
+
+    The ``default_context`` parameter was introduced in Symfony 6.2.
```
Comment (dbu) [inline reference/configuration/framework.rst]:
```suggestion
    The ``default_context`` parameter was introduced in Symfony 5.4.
```

File: serializer.rst [hunk 4]
```diff
@@ -130,14 +130,15 @@ configuration:
             serializer:
                 default_context:
                     enable_max_depth: true
+                    yaml_indentation: 2
```
Comment (dbu) [inline serializer.rst:133]: yaml_indentation however has been added in 6.2 only

File: reference/configuration/framework.rst (hunk 1 above)
Comment (OskarStark) [reply reference/configuration/framework.rst:2756]: Oh ok, in this case I would vote for adding the default context parameter against `5.4` branch, and the yaml stuff against `6.2`

Comment (dbu) [reply reference/configuration/framework.rst:2756]: right. i made https://github.com/symfony/symfony-docs/pull/17310

i will remove the changes from this MR, as they don't clash

Comment (dbu) [reply reference/configuration/framework.rst:2756]: actually the part about context builders was only added later, in 6.1. i will rebase this on 6.2 when you merged upstream, to keep the reference.

best would be if all the context settings would be documented, but i don't have time to do that.

Comment (OskarStark) [reply reference/configuration/framework.rst:2756]: I merged and unmerged your other PR, thanks!

**Comment by OskarStark** [comment]
Please also adjust the title of the PR after rebasing, thanks

File: reference/configuration/framework.rst (hunk 1 above)
Comment (dbu) [reply reference/configuration/framework.rst:2756]: thanks. rebased and solved the conflicts.

**Comment by dbu** [comment]
rebased, cleaned up and adjusted the title.

regarding the linter failure: looks like thats broken in all 6.x versions: https://github.com/symfony/symfony-docs/actions/runs/3160768594/jobs/5145596113

can you fix that in 6.0 and then merge it up to 6.1 and 6.2? or we just merge with the known error...

File: serializer.rst [hunk 5]
```diff
@@ -146,11 +147,13 @@ configuration:
...
         use Symfony\Config\FrameworkConfig;
         use Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer;
+        use Symfony\Component\Serializer\Encoder\YamlEncoder;
 
         return static function (FrameworkConfig $framework) {
             $framework->serializer()
                 ->defaultContext([
-                    AbstractObjectNormalizer::ENABLE_MAX_DEPTH => true
+                    AbstractObjectNormalizer::ENABLE_MAX_DEPTH => true,
+                    YamlEncoder::YAML_INDENTATION => 2,
```
Comment (dbu) [inline serializer.rst:156]: there is a build failure about this, but i don't understand it.

Comment (OskarStark) [reply serializer.rst:156]: any idea @Nyholm

File: reference/configuration/framework.rst [hunk 6]
```diff
@@ -2752,6 +2752,9 @@ A map with default context options that will be used with each ``serialize`` and
 call. This can be used for example to set the json encoding behavior by setting ``json_encode_options``
 to a `json_encode flags bitmask`_.
 
+You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
+to discover the available settings.
+
 .. versionadded:: 5.4
```
Comment (OskarStark) [inline reference/configuration/framework.rst]: Ro remove those messages, please rebase again, thanks

Comment (dbu) [reply reference/configuration/framework.rst]: thanks, rebased.

**Pull request merged**